src/core/sequence.ml

```diff
@@ -0,0 +1,41 @@
+
+type +'a node =
+   | Nil
+   | Cons of 'a * 'a t
+and 'a t = unit -> 'a node
+
+let empty () = Nil
+
+let return x () = Cons(x, empty)
+
+let rec map f i () = match i () with 
```

Comment:
I'm not fond of having `match` (or anything other than just `function` really) at the end of the line here. When I see a function declaration followed by a line starting with a pipe, I expect the last argument of the function to be implicitly pattern matched, ie. `= function`  instead of `= match ... with`